pkg/client/fake/client.go

```diff
@@ -1178,6 +1178,8 @@ func allowsUnconditionalUpdate(gvk schema.GroupVersionKind) bool {
 			return true
 		}
 	case "rbac":
```

Comment:
@vincepri The part I don't understand is why do we have to support both. There is no "rbac" API group in Kubernetes.

Is this to not break folks who are using this with Unstructured and are using a non-existing "rbac" api group for ClusterRole, etc.?

I would argue it's better to not continue to support a non-existing api group.

No objection though if you think it's the right thing to do. Would be good to add a godoc though why we are doing it

Reply:
I would agree
I don't know why it was "rbac" in the first place for this long, and how it has not been corrected before this

Reply:
I would just like to see this fixed either way

Reply:
Yup, no worries. Let's see what Vince thinks and then we get it merged one way or the other :)

Reply:
We can remove it, although we might want to mark this as a breaking change and add some release notes

Reply:
I have a slight tendency to remove "rbac" as it should be dead code and if it didn't than folks are doing pretty invalid stuff

Reply:
Thanks @sbueringer @vincepri,I've updated the PR